File: BikeShop.Backend/BikeShop.Payments/BikeShop.Payments.Domain/Entities/Discount.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BikeShop.Payments.Domain.Entities
{
    public class Discount : BaseEntity
    {
        public string SecretString { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Type { get; set; }
        public decimal Amount { get; set; }
        public decimal Limit { get; set; }
        public bool IsZeroLoss { get; set; } = true;
    }
}

File: BikeShop.Frontend/webclient/src/entities/DataTransferObjects/Discount.ts
export interface Discount {
    id: number
    createdAt: string
    updatedAt: string
    enabled: boolean
    secretString: string;
    name: string;
    description: string;
    type: string;
    target: string;
    bindToTargetId: number;
    amount: number;
    limit: number;
    isZeroLoss: boolean;
    isPersonal: boolean;
}